compiler/ballerina-parser/src/main/java/io/ballerina/compiler/internal/parser/BallerinaParser.java

```diff
@@ -12005,29 +12016,29 @@ private STNode parseIntermediateClause(boolean isRhsExpr, boolean allowActions)
             case CONFLICT_KEYWORD:
                 return null;
             default:
+                if (nextToken.kind == SyntaxKind.IDENTIFIER_TOKEN && nextToken instanceof STIdentifierToken) {
+                    if (((STIdentifierToken) nextToken).text.equals(SyntaxKind.COLLECT_KEYWORD.stringValue())) {
+                        return parseCollectClause(isRhsExpr);
+                    }
+                    if (((STIdentifierToken) nextToken).text.equals(SyntaxKind.GROUP_KEYWORD.stringValue())) {
+                        return parseGroupByClause(isRhsExpr);
+                    }
+                }
```

Comment:
With above suggestion we simplify this to,
```suggestion
                if (isKeywordMatch(SyntaxKind.COLLECT_KEYWORD, nextToken)) {
                    return parseCollectClause(isRhsExpr);
                }

                if (isKeywordMatch(SyntaxKind.GROUP_KEYWORD, nextToken)) {
                    return parseGroupByClause(isRhsExpr);
                }
```